pkg/config/analysis/analyzers/virtualservice/destinationrules.go

```diff
@@ -77,9 +80,10 @@ func (d *DestinationRuleAnalyzer) analyzeVirtualService(r *resource.Instance, ct
 	for _, ad := range getHTTPMirrorDestinations(vs) {
 		if !d.checkDestinationSubset(ns, ad.Destination, destHostsAndSubsets) {
 
-			m := msg.NewReferencedResourceNotFound(r, "mirror+subset in destinationrule",
-				fmt.Sprintf("%s+%s", ad.Destination.GetHost(), ad.Destination.GetSubset()))
-
+			m := msg.NewVirtualServiceDestinationMismatchInDestinationRule(
```

Comment:
Why do we need a new type?

Reply:
The same as above, it's adding new error message for the analyzer, and it's unclear to explain the mismatch between virtual service and destination rule, so I add a new type of analyzer message.

Reply:
I personally think this is not ideal. We do not want a new message for every single little issue, its useful to have higher level ones like `NewReferencedResourceNotFound`. Will leave to UX experts to decide though...